From 99d92261223ca965ee9198c432118c15c82401ea Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Fri, 25 Jun 2021 14:15:18 +0300
Subject: [PATCH] [SW-36106] gaudi2: update debug straps

updated the I2C/PCIe/DRAM debug strap bits as per Gaudi2 gpio generator
for both legacy and secure boot flow

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: Ie4414996d14817db2ff708ace0c3ce362f5dd6f7
---

diff --git a/arch/gaudi2/common.c b/arch/gaudi2/common.c
index 50ca153..94f987e 100644
--- a/arch/gaudi2/common.c
+++ b/arch/gaudi2/common.c
@@ -65,17 +65,14 @@
 	u32 strap_mask;
 
 	switch (func) {
-	case HL_DEBUG_HW_DBG_CARD:
-		strap_mask = BIT(11); /* Dump Select[3] */
-		break;
 	case HL_DEBUG_HW_PCIE:
-		strap_mask = BIT(10); /* Dump Select[2] */
+		strap_mask = BIT(12); /* Dump Select[4] */
 		break;
 	case HL_DEBUG_HW_DRAM:
-		strap_mask = BIT(8); /* Dump Select[0] */
+		strap_mask = BIT(11); /* Dump Select[3] */
 		break;
 	case HL_DEBUG_HW_I2C:
-		strap_mask = BIT(9); /* Dump Select[1] */
+		strap_mask = BIT(10); /* Dump Select[2] */
 		break;
 	default:
 		preboot_puts_hex("Unknown H/W function ", func, false);
diff --git a/arch/gaudi2_arc_gnu/common.c b/arch/gaudi2_arc_gnu/common.c
index f0ff998..61cc714 100644
--- a/arch/gaudi2_arc_gnu/common.c
+++ b/arch/gaudi2_arc_gnu/common.c
@@ -57,24 +57,20 @@
 	return mmPSOC_EFUSE_READ_CTRL;
 }
 
-/*TODO: Check this */
 bool hl_is_debug_mode(enum hl_debug_hw_func func)
 {
 	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L);
 	u32 strap_mask;
 
 	switch (func) {
-	case HL_DEBUG_HW_DBG_CARD:
-		strap_mask = BIT(11); /* Dump Select[3] */
-		break;
 	case HL_DEBUG_HW_PCIE:
-		strap_mask = BIT(10); /* Dump Select[2] */
+		strap_mask = BIT(12); /* Dump Select[4] */
 		break;
 	case HL_DEBUG_HW_DRAM:
-		strap_mask = BIT(8); /* Dump Select[0] */
+		strap_mask = BIT(11); /* Dump Select[3] */
 		break;
 	case HL_DEBUG_HW_I2C:
-		strap_mask = BIT(9); /* Dump Select[1] */
+		strap_mask = BIT(10); /* Dump Select[2] */
 		break;
 	default:
 		preboot_puts_hex("Unknown H/W function ", func, false);
